### PR TITLE
Shadow test accuracy

### LINKED ISSUE #19: Adding testing phase to shadow models training
# Issue
## Problem Description
Adding the testing phase to shadow model training will give us more info on our shadow models.

### LINKED ISSUE #157: Add shadow model test accuracy
# Feature
## Desired Behavior / Functionality
Implement method for shadow models to be able to give test/validation accuracy

## What Needs to Be Done
Copy target model test/validation method and create test dataset for each shadow model

## How Can It Be Tested
During shadow model training, it will logg test accuracy

### PR DESCRIPTION
**Pull request by johanos1**
# Description

Summary of changes

* Storing the test acc and test loss for the shadow models when trained in the ShadowModelSchema. The test data equals D_aux \ D_{shadow model}
* ShadowModelSchema updated

## Resolved Issues

- [ ] fixes #157 
- [ ] fixes #19 

## How Has This Been Tested?
- Unit tests pass
- Tested that the adult dataset works still

## Related Pull Requests

- #220 #235 must go in before this PR

### CODE REVIEW

File: leakpro/attacks/utils/shadow_model_handler.py [hunk 1]
```diff
@@ -165,17 +175,49 @@ def create_shadow_models(
...
+
+        Returns:
+        -------
+            Tuple[float, float]: The average accuracy and loss of the shadow models.
+
+        """
+        accuracy = 0.0
+        loss = 0.0
+        model.eval()
+        model.to(self.device)
```
Comment (fazelehh) [inline leakpro/attacks/utils/shadow_model_handler.py:209]: use no_grad for efficiency

File: examples/mia/tabular_mia/utils/adult_model_preparation.py [hunk 2]
```diff
@@ -61,7 +61,7 @@ def create_trained_model_and_metadata(model, train_loader, test_loader, epochs =
             output = model(data)
 
             loss = criterion(output, target)
-            pred = sigmoid(output) >= 0.5
+            pred = output >= 0.5
```
Comment (henrikfo) [inline examples/mia/tabular_mia/utils/adult_model_preparation.py:64]: The sigmoid functionality is instead baked in with the model?

File: leakpro/attacks/utils/shadow_model_handler.py (hunk 1 above)
Comment (henrikfo) [reply leakpro/attacks/utils/shadow_model_handler.py:209]: +1

File: examples/mia/tabular_mia/utils/adult_model_preparation.py (hunk 2 above)
Comment (johanos1) [reply examples/mia/tabular_mia/utils/adult_model_preparation.py:64]: yea, we are using `nn.BCEWithLogitsLoss()`
